### DISCUSSION TITLE
LVT ISMN reader questions

### DISCUSSION THREAD

**Discussion by jeylander** [NASA-LIS/LISF#1269] [General]
I'm trying to use LVT to compare soil moisture observations from the ISMN data stream to LIS output.  I opened up the ISMN reader code in LVT and notice that the code creates a file listing by searching through all $ISMNDIR/$FYR/ and looking for files with *sm*stm in the filename (where $represents variables in the fortran code).   Since the ISMN data file structure is arranged very differently (by observation network, then station name), I'm assuming whomever wrote the ISMN reader had downloaded all the data from multiple observation networks and then combined all the *sm*stm files into a yearly directory.  Is that a correct assumption, or was there additional processing done to combine only specific observations from the ISMN network, or did I grab the incorrect format?  (I believe the reader requires CEOP formatted data).  Thanks in advance.

John

**Comment by dmocko** [comment]
Hi @jeylander 

You are correct, the LVT reader looks for files with `sm` in the filename and with the suffix `stm`:

``ls_comm = 'ls '//trim(odir)//'/'//trim(fyr)//'/*sm*stm > file_list'``

Yes, we moved all files into a directory structure by year.  There was not any additional processing performed.

e.g, move this file: `ARM/Ashton/ARM_ARM_Ashton_sm_0.050000_0.050000_Hydraprobe-II-Sdi-12-E_20210101_20211231.stm`
into this directory: `ISMN/2021`

A couple other notes:

1. It's better to have a symbolic link in your LVT running directory directly to your ISMN directory, instead of a long path:

In your lvt.config: `ISMN observation directory:             ./ISMN`
In your LVT running directory: `lrwxrwxrwx 1 dmocko s1189       22 May 26  2020 ISMN -> /[long_path_to_ISMN_directory]`

You probably need to do this, as the ISMN filenames are very long, and with a long directory as well, LVT may throw an error due to the entire path to the file being too long.

2. You may need to add this to your job script, as LVT will open all files for a single year.  Otherwise, LVT may cause an error:

```
ulimit -s unlimited
ulimit -n 4096
```

**Reply by jeylander** [reply]
awesome.  Thank you David!
